effectful/ops/syntax.py

```diff
@@ -180,10 +181,17 @@ def _(t: Callable[P, T], *, name: Optional[str] = None) -> Operation[P, T]:
 
 @defop.register(Operation)
 def _(t: Operation[P, T], *, name: Optional[str] = None) -> Operation[P, T]:
+
+    @functools.wraps(t)
     def func(*args, **kwargs):
         raise NotImplementedError
 
-    functools.update_wrapper(func, t)
+    if name is None:
+        name = (
+            getattr(t, "__name__", str(t))[:10000]
```

Comment:
Why not take the whole name? If length is a concern, 10000 chars is definitely already too long.

Reply:
10000 is an arbitrary upper bound to keep the names generated by `defop` from growing without limit

Reply:
Ah I see the concern. I think it would be better to keep the freshness id separate from the string name and regenerate the id when we copy an operation.

Reply:
I've found it useful in practice when debugging to have the name reflect the sequence of `defop`s that led to an `Operation`. I think it's fine as long as names have a bounded length and don't affect `Operation` equality semantics.

Reply:
Ok, that's fine.